vignettes/questionnaires.Rmd

```diff
@@ -0,0 +1,627 @@
+  68L, "GDS02-Prefer to Stay Home", "GDS0209", "VISIT 4", 4, "2013-02-13", "NO", 0L,
+  69L, "GDS02-Memory Problems", "GDS0210", "VISIT 4", 4, "2013-02-13", "NO", 0L,
+  70L, "GDS02-Wonderful to Be Alive", "GDS0211", "VISIT 4", 4, "2013-02-13", "NO", 1L,
+  71L, "GDS02-Feel Worthless", "GDS0212", "VISIT 4", 4, "2013-02-13", "NO", 0L,
+  72L, "GDS02-Feel Full of Energy", "GDS0213", "VISIT 4", 4, "2013-02-13", "YES", 0L,
+  73L, "GDS02-Feel Hopeless", "GDS0214", "VISIT 4", 4, "2013-02-13", "NO", 0L,
+  74L, "GDS02-Most People Better Off Than You", "GDS0215", "VISIT 4", 4, "2013-02-13", "NO", 0L
+) %>%
+  mutate(QSCAT = "GDS SHORT FORM")
+
+qs <- bind_rows(qs_gad7, qs_gdssf) %>%
+  mutate(
+    STUDYID = "STUDYX",
+    USUBJID = "P0001",
+    .before = everything()
+  )
+
+dataset_vignette(qs)
+
+adsl <- tibble::tribble(
+  ~STUDYID, ~USUBJID, ~SITEID, ~ITTFL,    ~TRTP, ~TRTSDT,                      ~DTHCAUS,
+  "STUDYX",  "P0001",     13L,    "Y", "DRUG A", lubridate::ymd("2012-11-16"), NA_character_
+)
+```
+# Original Items
+
+The original items can be handled in the same way like in a [BDS finding ADaM](bds_finding.html). For example:
+
+```{r eval=TRUE}
+adqs <- qs %>%
+  # add ADSL variables
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = exprs(TRTSDT, DTHCAUS),
+    by_vars = exprs(STUDYID, USUBJID)
+  ) %>%
+  # add analysis parameter variables
+  mutate(
+    PARAMCD = QSTESTCD,
+    PARAM = QSTEST,
+    PARCAT1 = QSCAT,
+    AVALC = QSORRES,
+    AVAL = QSSTRESN
+  ) %>%
+  # add timing variables
+  derive_vars_dt(new_vars_prefix = "A", dtc = QSDTC) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  mutate(
+    AVISIT = VISIT,
+    AVISITN = VISITNUM
+  )
+```
+
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adqs, USUBJID, PARCAT1, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, PARCAT1, AVALC, AVAL, ADY, AVISIT)
+)
+```
+
+We handle unscheduled visits as normal visits. For deriving visits based on
+time-windows see [Visit and Period Variables](visits_periods.html#visits) and
+for flagging values to be used for analysis `derive_var_extreme_flag()`.
+
+# Scales and Scores
+
+Scales and Scores are often derived as the sum or the average across a subset of
+the items. For the GAD-7 questionnaire the total score is derived as the sum.
+The `derive_summary_records()` function with `sum()` can be used to derive it as
+a new parameter. In the example we derive a separate ADaM dataset for each
+questionnaire. Depending on the analysis needs it is also possible that an ADaM
+contains more than one questionnaire or all questionnaires.
+```{r eval=TRUE}
+adgad7 <- adqs %>%
+  filter(PARCAT1 == "GAD-7 V2") %>%
+  derive_summary_records(
+    by_vars = exprs(USUBJID, AVISIT, ADT, ADY),
+    analysis_var = AVAL,
+    summary_fun = function(x) sum(x, na.rm = TRUE),
+    filter = str_detect(PARAMCD, "GAD020[1-7]"),
```

Comment:
ohhh - can we give a tip that we use regex to help with filtering?

Reply:
I see it is used later as well, maybe just one mention suffices but this is really nice trick to have in programmer back pocket :)

Reply:
Tip added